Parsers/ASimNetworkSession/ProductParsers/ASimNetworkSessionPaloAltoPanOS.yaml

```diff
@@ -0,0 +1,118 @@
+Parser:
+  Title: Palo Alto Network Sessions
+  Version: '0.1'
+  LastUpdated: Dec 20, 2021
+Product:
+  Name: PaloAlto
+Normalization:
+  Schema: NetworkSessions
+  Version: '0.2.0'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+  This Query Parser maps Palo Alto PanOS Firewall Events (CommonSecurityLogs) to the Advanced SIEM Information Model Network Session schema.
+ParserName: ASimNetworkSessionPaloAltoPanOS
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let NWParser=(disabled:bool=false){
+  CommonSecurityLog | where not(disabled)
+  | where DeviceVendor=="Palo Alto Networks" and (Activity=="TRAFFIC")
+  | parse AdditionalExtensions with "PanOSDstPackets="DstPackets:long";PanOSsrc_packets="SrcPackets:long";start="EventStartTime:datetime";reason="*
+  | project-rename 
+        EventVendor=DeviceVendor // Not documented
+      , EventProduct=DeviceProduct // Not Documented
+      , EventProductVersion=DeviceVersion // Not Documented
+      , DvcHostname=DeviceName   
+      , EventResourceId=_ResourceId  // Not Documented 
+      , NetworkApplicationProtocol=ApplicationProtocol
+      , SrcZone=DeviceCustomString4 
+      , DstZone=DeviceCustomString5 
+      , NetworkRuleName=DeviceCustomString1
+      , NetworkProtocol=Protocol
+      , SrcBytes=SentBytes
+      , DstBytes=ReceivedBytes 
+      , SrcUsername=SourceUserID 
+      , DstUsername=DestinationUserID 
+      , EventSeverity=LogSeverity // not documented
+      , NetworkPackets=DeviceCustomNumber2 
+      , SrcNatIpAddr=SourceTranslatedAddress
+      , DstNatIpAddr=DestinationTranslatedAddress
+      , NetworkConnectionHistory=FlexString1 // Flags
+      // , UrlCategory=DeviceCustomString2 -- websessions
+      // , LogProfile=DeviceCustomString6  
+      //, VirtualSystem=DeviceCustomString3
+    // Proxy
+    // , UrlOriginal=RequestURL
+    // , HttpUserAgentOriginal=RequestClientApplication
+    // , HttpContentType=RequestContext
+    // , HttpRequestMethod=RequestMethod
+  | extend
+      NetworkBytes=tolong(FlexNumber1)
+      , SrcUsernameType="UPN"
+      , DstUsernameType="UPN"
+      , TimeGenerated  
+      , EventType="NetworkSession" // it is not clear from doc when NWConnection and when NWSession
+      , EventCount=toint(1)
+      , EventResult=case(DeviceAction=="allow","Success","Failure")
+      , NetworkSessionId=tostring(DeviceCustomNumber1)
+      , NetworkDuration=DeviceCustomNumber3
+      , EventSchemaVersion="0.2.0"
+      , EventSchema="imNetworkSession"
+      , DvcAction=case(DeviceAction=="reset client", "Reset Source",  DeviceAction=="reset server", "Reset Destination",  DeviceAction=="reset both", "Reset",  DeviceAction)
```

Comment:
DvcAction is normalized, so it can't take DeviceAction as a default. We should research to understand all possible values (this was my blocker for fast translation to V2).

Also, store the original value in DvcOriginalAction

Reply:
Replaced with lookup